### ISSUE TITLE
Allow folderfirst option to use folderrec

### ISSUE THREAD

**Issue by jayhasgin** [imapsync/imapsync#356]
Presently, `folderfirst` doesn't check `folderrec`, and there is no option to include all sub-folders for each folder mentioned through `folderfirst`. I've tested following code to `sort_requested_folders` subroutine to allow `folderfirst` use `folderrec` option, and it works great.

```
...
sub sort_requested_folders
{
        my @requested_folders_sorted = () ;
        
        if( scalar @folderrec && scalar @folderfirst && scalar @h1_folders_all ) {
		my @folderfirstrec = ();
		foreach my $folderfirst ( @folderfirst ) {
			push @folderfirstrec, $folderfirst ;
			if ( grep ( /^$folderfirst$/ , @folderrec) ) {
				push ( @folderfirstrec, ( grep { /^$folderfirst\.(.+)$/ } @h1_folders_all ) );
			}
		}
		@folderfirst = @folderfirstrec;
        }

        $sync->{ debug } and myprint "folderfirst: @folderfirst\n" ;
...
```

The same can also be added for `folderlast`. Having this check is very useful, esp. when folders mentioned using `folderfirst` option has new sub-folders, and also helps in preventing duplicates.

**Comment by gilleslamiral** [comment]
This is a functional change so I can't allow this.
I could enable it as a --folderfirstrec option. 
Do not presume that the dot is always the separator.

**Comment by jayhasgin** [comment]
You're right! I was only giving an example based on my use case, I'm no expert in mailboxes or in Perl coding. A `--folderfirstrec` option with regex input would be great. Thank you!

**Comment by gilleslamiral** [comment]
I put this  `--folderfirstrec` suggestion in the TODO file.
https://imapsync.lamiral.info/TODO
